tests/unit/adapters/drf-test.js

```diff
@@ -33,6 +33,7 @@ test('buildURL', function(assert) {
   var adapter = this.subject();
   assert.equal(adapter.buildURL('Animal', 5, null), 'test-host/test-api/animals/5/');
   assert.equal(adapter.buildURL('FurryAnimals', 5, null), 'test-host/test-api/furry-animals/5/');
+  assert.equal(adapter.buildURL('Animal', null, null, 'query', { limit: 10 }), 'test-host/test-api/animals/');
```

Comment:
I think the idea for `urlForQuery` is 'build a base URL for a query request'. I think we just need to allow users to override `buildURL` to adjust the base URL based on the data in any of the parameters (including query data) in case they need this. For example maybe I need the base URL (i.e. API resource) to change when doing a specific query request. Just my opinion / guess though.